src/main/java/com/palantir/gradle/gitversion/VersionDetails.java

```diff
@@ -59,7 +59,23 @@ private String expensiveComputeRawDescription() {
             return null;
         }
 
-        return new JGitDescribe(git).describe(args.getPrefix());
+        String nativeGitDescribe = new NativeGitDescribe(git.getRepository().getDirectory())
+                .describe(args.getPrefix());
+        String jgitDescribe = new JGitDescribe(git)
+                .describe(args.getPrefix());
+
+        // If native failed, return JGit one
+        if (nativeGitDescribe == null) {
+            return jgitDescribe;
+        }
+
+        // If native succeeded, make sure it's same as JGit one
+        if (!nativeGitDescribe.equals(jgitDescribe)) {
+            throw new IllegalStateException(String.format("Inconsistent git describe: native was %s and jgit was %s. " +
+                    "Please report this on github.com/palantir/gradle-git-version", nativeGitDescribe, jgitDescribe));
+        }
```

Comment:
If we're saying we don't need the jgit stuff, then why are we still doing the dual implementation?

Reply:
I am not saying that. I'm in favour of reverting this to the way 0.11.0 behaved because that's the last non-rc and it's being used in a lot of places.
We should leave it as it is even if it's using a hybrid of jgit and git, and only attempt to fix it if we encounter problems.